<launch>
  <arg name="mav_name" default="firefly"/>
  <arg name="frame_id" default="odom" />
  <arg name="robot_radius" default="0.72" />
  <arg name="voxel_size" default="0.1" /> <!--"0.2" is the default-->
  <!--arg name="voxblox_path" default="/home/emilk/Downloads/mav_voxblox_planning_maps/machine_hall/voxblox/rs_esdf_0.10.voxblox" /-->

  <group ns="$(arg mav_name)">
    <node name="voxblox_rrt_planner" pkg="voxblox_rrt_planner" type="voxblox_rrt_planner_node" output="screen" clear_params="true">
      <remap from="odometry" to="ground_truth/odometry" />
      <remap from="voxblox_rrt_planner/esdf_map_in" to="esdf_map" />
      <remap from="voxblox_rrt_planner/tsdf_map_in" to="tsdf_map" />
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="16" />

      <param name="visualize" value="true" />
      <param name="robot_radius" value="$(arg robot_radius)" />

      <!--param name="voxblox_path" value="$(arg voxblox_path)" /-->
      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="num_seconds_to_plan" value="2.0" />
      <param name="do_smoothing" value="false" />
      <param name="frame_id" value="$(arg frame_id)" />
      <param name="world_frame" value="$(arg frame_id)" />
      <param name="trust_approx_solution" value="true" />
      <param name="publish_traversable" value="true" />
      <param name="split_at_collisions" value="true" />
      <param name="optimize_time" value="true" />
      <param name="verbose" value="true" />
      <param name="simplify_solution" value="false" />
      <param name="optimistic" value="true" />

      <!-- Set maximum distance between nodes in the tree -->
      <param name="max_branch_length" value="5.0" />

      <!-- Specify if constant altitude flight is required -->
      <param name="constant_altitude" value="true" />

      <!-- Specify planner to use from the following: -->
      <!-- RRT connect = kRrtConnect -->
      <!-- RRT* = kRrtStar -->
      <!-- InformedRRT* = kInformedRrtStar -->
      <!-- Probabilistic RoadMap = kPrm -->
      <!-- Bit* = kBitStar -->
      <param name="planner_type" value="kInformedRrtStar" />

      <!-- Activate replanning functionality -->
      <param name="activate_replanning" value="true" />

      <!-- Timer period for global trajectory replanning -->
      <param name="gp_replan_dt" value="1.0" />

      <!-- Time window for moving ahead along the global path -->
      <!-- Should satisfy  gp_replan_lookahead_sec_>=num_seconds_to_plan-->
      <param name="gp_replan_lookahead_sec" value="2.0" />

      <!-- WEIGHTS FOR OPTIMIZATION OBJECTIVE of the form (alpha*LenghtObj + beta*AltitudeObj) -->
      <param name="alpha" value="1.0" />
      <param name="beta" value="10.0" />

    </node>
  </group>
</launch>
